phoenix-core/src/it/java/org/apache/phoenix/jdbc/FailoverPhoenixConnection2IT.java

```diff
@@ -470,6 +474,75 @@ public void testAllWrappedConnectionsNotClosedAfterStandbyURLChange() throws Exc
         }
     }
 
+    /**
+     * Test connections behaviour when doing 2-step failover.
+     * cluster 1 ( ACTIVE --> ACTIVE_TO_STANDBY --> STANDBY )
+     */
+    @Test(timeout = 300000)
+    public void testAllWrappedConnectionsClosedAtRightTimeDuringClusterRoleChange()
+            throws Exception {
+        //Write Some data to ACTIVE cluster
+        Connection connection = createFailoverConnection();
+        Statement statement = connection.createStatement();
+        short numOfRows = 100;
+        for (int i = 0; i < numOfRows; i++) {
+            statement.executeUpdate(String.format("UPSERT INTO %s VALUES(%d, 1984)", tableName, i));
+        }
+        connection.commit();
+
+        //Creating some connections to ACTIVE cluster
+        short numberOfConnections = 10;
+        //Create FailoverPhoenixConnections with default urls
+        List<Connection> connectionList = new ArrayList<>(numberOfConnections);
+        for (short i = 0; i < numberOfConnections; i++) {
+            connectionList.add(createFailoverConnection());
+        }
+
+        //Transit Role1 ACTIVE --> ACTIVE_TO_STANDBY
```

Comment:
I would just add one more test case: Check when  we transition a cluster from ATS --> Active. It should not close any connections in this case also.

Reply:
Thanks, I @shahrs87 I have added one more test for early rollback Active --> ATS --> Active which should not affect connections at all